tox.ini

```
@@ -1,40 +1,41 @@
 [tox]
 envlist =
-    cov-clean
-    cov-combine
-    cov-report
+    lint
+    mypy
+    test-lazy-imports
+    coverage_clean
     py{311,310,39,38,37,36}
     py36-mindeps
+    coverage_report
 skip_missing_interpreters = true
-minversion = 3.0.0
+# TODO: set a minversion after we drop support for python3.6
+# minversion = 4.0.0
 
 [testenv]
-usedevelop = true
+# build a wheel, not a tarball, and use a common env to do it (so that the wheel is shared)
+package = wheel
+wheel_build_env = build_wheel
+
 extras = dev
 deps =
     mindeps: requests==2.19.1
     mindeps: pyjwt==2.0.0
     mindeps: cryptography==3.3.1
     mindeps: typing_extensions==4.0
-commands = coverage run -m pytest {posargs}
+commands = coverage run --source globus_sdk -m pytest {posargs}
 depends =
-    py36,py37,py38,py39,py36-mindeps: cov-clean
-    cov-combine: py36,py37,py38,py39,py36-mindeps
-    cov-report: cov-combine
+    py{36,37,38,39,310,311,36-mindeps}: coverage_clean
+    coverage_report: py{36,37,38,39,310,311,36-mindeps}
```

I think that the `-mindeps` marker can be left off of this, since it's a marker...but if not, then that may finally explain some challenges I've had in another codebase (and had thought were issues related to PyPy being slow)!

Does this work?

```suggestion
    py{36,37,38,39,310,311}: coverage_clean
    coverage_report: py{36,37,38,39,310,311}
```

---

I don't really know on this one, since it seems like a detail of how `depends` evaluates. Let me toy with it a bit.

---

If it _is_ needed then this will finally, _finally_ explain a problem I've had in another repo. I'm actually going to test that now. If it resolves a problem I've had with PyPy tests producing coverage files after coverage combine has already run, then it will indicate that tox requires explicit lists, and all markers must be provided!

---

It is absolutely necessary. This knowledge fixes a problem I've had in another repo, too.

The moral of the story is: "All tox environment dependencies must be specified in full, including their markers."